packages/@n8n/api-types/src/dto/api-keys/create-or-update-api-key-request.dto.ts

```diff
@@ -8,6 +8,16 @@ const xssCheck = (value: string) =>
 		whiteList: {},
 	});
 
+const expirationDateCheck = (value?: number) => {
+	if (!value) return true;
+	return value > Date.now() / 1000;
+};
+
 export class CreateOrUpdateApiKeyRequestDto extends Z.class({
 	label: z.string().max(50).min(1).refine(xssCheck),
+	/** expirationUnixTimestamp is ignored when updating */
+	expirationUnixTimestamp: z
+		.number()
+		.optional()
```

Comment:
We accept `number | undefined` here. But the type for `expiresAt` in `ApiKey` uses `number | null`. Could we use the same for both?

e.g. change this to 
```suggestion
		.nullable()
```

or change `expiresAt` to `expiresAt?: number`

Reply:
This a big confusing. Maybe we should call this also `expiresAt` and with the documentation people should know the format (unix timestamp in seconds). Also, maybe create a different DTO for the update operation, since when updating you can do the same as when creating except for updating the expiration. The this field will be set to null when the user selected no expiration, which makes more clear, as that is the value the `expiresAt` takes when the API key has no expiration.

Reply:
Oh right. In that case we can't use the same DTO for create and update, if they don't have the same fields. Better to separate them. I think `expiresAt` is also enough, as long as we document what it means 👍